src/libs/Performance.js

```diff
@@ -25,43 +33,155 @@ function diffObject(object, base) {
-            } else {
-                performance.measure('runJsBundle', 'runJsBundleStart', 'runJsBundleEnd');
-            }
-        }
-    }).observe({type: 'react-native-mark', buffered: true});
-}
+if (canCapturePerformanceMetrics()) {
+    /**
+     * Sets up an observer to capture events recorded in the native layer before the app fully initializes.
+     */
+    setupPerformanceObserver = () => {
+        const performanceReported = require('react-native-performance-flipper-reporter');
+        performanceReported.setupDefaultFlipperReporter();
 
-/**
- * Outputs performance stats. We alert these so that they are easy to access in release builds.
- */
-function printPerformanceMetrics() {
-    const performance = require('react-native-performance').default;
-    const entries = _.map(performance.getEntriesByType('measure'), entry => ({
-        name: entry.name, duration: Math.floor(entry.duration),
-    }));
-    alert(JSON.stringify(entries, null, 4));
+        const perfModule = require('react-native-performance');
+        perfModule.setResourceLoggingEnabled(true);
+        performance = perfModule.default;
+
+        // Monitor some native marks that we want to put on the timeline
+        new perfModule.PerformanceObserver((list, observer) => {
+            list.getEntries()
+                .forEach((entry) => {
+                    if (entry.name === 'nativeLaunchEnd') {
+                        performance.measure('nativeLaunch', 'nativeLaunchStart', 'nativeLaunchEnd');
+                    }
+                    if (entry.name === 'downloadEnd') {
+                        performance.measure('jsBundleDownload', 'downloadStart', 'downloadEnd');
+                    }
+                    if (entry.name === 'runJsBundleEnd') {
+                        performance.measure('runJsBundle', 'runJsBundleStart', 'runJsBundleEnd');
+                    }
+
+                    // We don't need to keep the observer past this point
+                    if (entry.name === 'runJsBundleEnd' || entry.name === 'downloadEnd') {
+                        observer.disconnect();
+                    }
+                });
+        }).observe({type: 'react-native-mark', buffered: true});
+
+        new perfModule.PerformanceObserver((list) => {
+            list.getEntriesByType('mark')
+                .forEach((mark) => {
+                    if (mark.name.endsWith('_end')) {
+                        const end = mark.name;
+                        const name = end.replace(/_end$/, '');
+                        const start = `${name}_start`;
+                        performance.measure(name, start, end);
+                    }
+
+                    if (mark.name === `${CONST.TIMING.SIDEBAR_LOADED}_end`) {
+                        performance.measure('TTI', 'nativeLaunchStart', mark.name);
+                        printPerformanceMetrics();
+                    }
+                });
+        }).observe({type: 'mark', buffered: true});
```

Comment:
Forgot to document this
It's based on this comment: https://github.com/Expensify/App/issues/4549#issuecomment-900223536

Will update with docs tomorrow